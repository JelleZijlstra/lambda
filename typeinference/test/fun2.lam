let f = \z. z.l2 z.l1 in
let r1 = {
    l1 = 42,
    l2 = \x. x = 0,
    l3 = true
} in
let r2 = {
	l1 = 3,
	l2 = \x. x + 1,
	l3 = \x. x
} in
(f r1, (f r2, f))
