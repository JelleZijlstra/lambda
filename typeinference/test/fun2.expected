%%stdout%%
Result: module
	let __result__ = (false, (4, \z. z.l2 z.l1))
end
%%stderr%%
